instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingPredicate.java

```diff
@@ -0,0 +1,43 @@
+package brave.kafka.streams;
+
+import brave.Span;
+import brave.Tracer;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Predicate;
+
+class TracingPredicate<K, V> extends AbstractTracingTransformer<K, V, KeyValue<K, V>> {
+
+  final KafkaStreamsTracing kafkaStreamsTracing;
+  final String spanName;
+  final Predicate<K, V> delegatePredicate;
+
+  TracingPredicate(KafkaStreamsTracing tracing, String spanName,
+      Predicate<K, V> delegatePredicate) {
+    this.kafkaStreamsTracing = tracing;
+    this.spanName = spanName;
+    this.delegatePredicate = delegatePredicate;
+  }
+
+  @Override
+  public KeyValue<K, V> transform(K key, V value) {
+    Span span = kafkaStreamsTracing.tracing.tracer().nextSpan();
```

Comment:
@jeqo initially i was using `kafkaStreamsTracing.nextSpan(processorContext)` here but Kafka did not like it. `AbstractProcessorContext` barfed with `throw new IllegalStateException("This should not happen as headers() should only be called while a record is processed");` . Can you shed more light on this ?

Reply:
Take a look to https://github.com/openzipkin/brave/blob/master/instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingTransformer.java#L33. Instrumentation need to access headers, and headers are part of the `processorContext`, initialized on `init`.